libraries/Microsoft.Bot.Builder/BotState.cs

```diff
@@ -6,34 +6,104 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Bot.Builder
 {
+
+    /** 
+     * State information cached off the context object by a `BotState` instance.
+     */
+    internal class CachedBotState
+    {
```

Comment:
Our coding standard is one class per file. This is enforced by StyleCop, which we should really get running on your machine.